### ISSUE TITLE
Glob pattern does not recurse into Magento subdirectories when assembling schema files

### ISSUE THREAD

**Issue by nwcasebolt** [JetBrains/js-graphql-intellij-plugin#706]
**Describe the bug**

Magento 2.4.6-p2 defines its own directives (such as `@doc` and `@resolver`) at `vendor/magento/module-graph-ql/etc/schema.graphqls`, but custom project files in `app/code` show errors on these directives. When I copy Magento's schema file to `vendor/magento`, these errors resolve. If I place the file in any lower subdirectory, the errors return. The plugin seems unable to assemble `schema.graphqls` files from subdirectories lower than `vendor/magento/`. This obtains whether I rely on the plugin's OOB implementation or explicitly configure a [basic configuration](https://github.com/JetBrains/js-graphql-intellij-plugin/blob/46adefce38fa6bde391d9f50e888504e3a38f516/README.md?plain=1#L117-L132) or [composite schema](https://github.com/JetBrains/js-graphql-intellij-plugin/blob/46adefce38fa6bde391d9f50e888504e3a38f516/README.md?plain=1#L134-L155).

**To Reproduce**

1. Create a root project directory. Create an `app/code` directory and a `vendor/magento/module-graph-ql` directory.
2. Clone an example Magento 2 module into `app/code/`. [MagePlaza's SMTP-2 module](https://github.com/mageplaza/magento-2-smtp) will work.
3. Copy [Magento's GraphQL module](https://github.com/magento/magento2/tree/2.4-develop/app/code/Magento/GraphQl) into `vendor/magento/quote-graph-ql`.
4. In PhpStorm, open MagePlaza's schema.graphqls file in `etc` and observe errors on Magento's directives.
5. Move or copy Magento's schema.graphqls file in `etc` to the `vendor/magento` directory.
6. Observe that there are no further directive errors in MagePlaza's schema file.

**Expected behavior**
The plugin recurses into _all_ project subdirectories and assembles _all_ matching schema files.

**Version and Environment Details**
Operation system: macOS Sonoma 14.2.1
IDE name and version: PhpStorm 2023.3.4
Plugin version: 233.13135.65

**Comment by vepanimas** [comment]
Hi @nwcasebolt! You mentioned a glob pattern in the issue title, but I don't see any configs in the provided code or in steps to reproduce. Could you please elaborate?

**Comment by nwcasebolt** [comment]
> Hi @nwcasebolt! You mentioned a glob pattern in the issue title, but I don't see any configs in the provided code or in steps to reproduce. Could you please elaborate?

Certainly. I see how my title is misleading, since it refers to a specific part of the problem I have rather than the holistic issue. I've modified the title and updated my reproduction steps to be, I hope, more clear.

**Comment by vepanimas** [comment]
@nwcasebolt Thank you for the clarification! If you stop just after the first step after downloading all the corresponding projects and add no configs at all it will work out of the box. If you create a configuration, I still need to see its content and know the place where you put it, it's important.

For example, if you put such `graphql.config.yml` in the content root, all directives will be resolved:
```
schema:
  - app/code/etc/schema.graphqls
  - vendor/magento/module-graph-ql/etc/schema.graphqls
documents: '**/*.graphql'
```

If you don't want to specify exact paths, this also will work:

```
schema: '**/*.graphqls'
documents: '**/*.graphql'
```

or to include not only `.graphqls`, but also regular `.graphql` files:

```
schema: '**/*.graphql*'
documents: '**/*.graphql'
```
